proposals/lip-0047.md

```diff
@@ -498,11 +498,147 @@ This module does not define any specific logic for other modules.
+  * Check that the `name` property of every entry `validator` in the `genesisBlockAssetObject.validatorSubstore.validators` array contains only characters from the set `[a-z0-9!@$&_.]`, is not empty and is at most `MAX_LENGTH_NAME` characters long.
+  * Check that the `generatorKey` property of every entry `validator` in the `genesisBlockAssetObject.validatorSubstore.validators` array has length 32.
+  * Check that the `blsKey` property of every entry `validator` in the `genesisBlockAssetObject.validatorSubstore.validators` array has length 48.
+  * Check that the `proofOfPossession` property of every entry `validator` in the `genesisBlockAssetObject.validatorSubstore.validators` array has length 96.
+
+* Checks for the `genesisBlockAssetObject.snapshotSubstore`: 
+  * The entries in the `genesisBlockAssetObject.snapshotSubstore.activeValidators` array are ordered lexicographically according to `address`.
+  * For every element `activeValidator` in the `genesisBlockAssetObject.snapshotSubstore.activeValidators` array, there is an entry with `address == activeValidator.address` in the `genesisBlockAssetObject.validatorSubstore.validators` array.
+  * The `weight` property of every element `activeValidator` in the `genesisBlockAssetObject.snapshotSubstore.activeValidators` array is a positive integer.
+  * Let `totalWeight` be the sum of the `weight` property of every element `activeValidator`. Then `totalWeight` has to be less than or equal to `MAX_UINT64`.
+  * The value of `genesisBlockAssetObject.snapshotSubstore.threshold` is within the following range:
+     * Minimum value: ⌊ ⅓ × `totalWeight`⌋ + 1
+     * Maximum value: `totalWeight`
+
+    where ⌊⋅⌋ is the floor function.
+
+* Create an entry in the validator substore for each entry `validator` in the `genesisBlockAssetObject.validatorSubstore.validators` array as:
+  * `storeKey`: `validator.address`.
+  * The `storeValue` is the serialization of the object `validatorObject` following `validatorObjectSchema` with `validatorObject.name = validator.name`. 	
+
+* Create an entry in the name substore for each entry `validator` in the `genesisBlockAssetObject.validatorSubstore.validators` array as:
+  * `storeKey`: `validator.name` serialized as a utf-8 encoded string.
+  * The `storeValue` is the serialization of the object `validatorAddress` following `validatorAddressSchema` with `validatorAddress.address = validator.address`.
+
+* Create three entries in the snapshot substore as:
+  * The `storeKey` of each of the entries are `uint32be(0)`, `uint32be(1)`, and `uint32be(2)` respectively.
+  * The `storeValue` is the same for the three entries and equal to the serialization of `genesisBlockAssetObject.snapshotSubstore`.
+
+* Create an entry in the chain properties substore as specified above.
+  * The `storeValue` is the serialization of an object following `chainPropSchema` where all the properties are initialized to their default values. 
```

Comment:
Above you sate about the `roundEndHeight`:
> Its initial value is set after the execution of the genesis block.

I think this should still be true and we should not assume that the genesis height is `0`. 

I would also remove the comment about the initialization that you have in the section above as it is redundant there.

Reply:
Ah, now I see that this value is overwritten later. It is normally not good practice to initialize something in between with an incorrect value, even if it is not used. 

Maybe consider setting it directly to the final value or first to the genesis height and then doing
```
    chainProperties.roundEndHeight = chainProperties.roundEndHeight + len(snapshotStore(0).validators)
```
below.

Reply:
changed to the second suggestion